main.go

```diff
@@ -355,6 +359,9 @@ func clientValidateFlags() error {
 	if err := validateCipherSuites(); err != nil {
 		return err
 	}
+	if *statusTargetAddress != "" {
+		return fmt.Errorf("--status-target should not be set in client mode")
```

Comment:
Instead of doing `app.Flag`, use `serverCommand.Flag` (there's examples above with other flags). That way the flag can't be set in client mode and you can drop the check here.